Source/SwiftLintFramework/Rules/NumberSeparatorExamples.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  NumberSeparatorExamples.swift
+//  SwiftLint
+//
+//  Created by Marcelo Fabri on 12/29/16.
+//  Copyright © 2016 Realm. All rights reserved.
+//
+
+import Foundation
+
+internal struct NumberSeparatorExamples {
+
+    static func nonTriggeringExamples() -> [String] {
+        return ["-", "+", ""].flatMap { (sign: String) -> [String] in
+            [
+                "let foo = \(sign)100",
+                "let foo = \(sign)1_000",
+                "let foo = \(sign)1_000_000",
+                "let foo = \(sign)1.000_1",
+                "let foo = \(sign)1_000_000.000_000_1",
+                "let binary = \(sign)0b10000",
+                "let binary = \(sign)0b1000_0001",
+                "let hex = \(sign)0xA",
+                "let hex = \(sign)0xAA_BB",
+                "let octal = \(sign)0o21",
+                "let octal = \(sign)0o21_1",
+                "let exp = \(sign)1_000_000.000_000e2"
+            ]
+        }
+    }
+
+    private static func triggeringExamples(signs: [String]) -> [String] {
+        return signs.flatMap { (sign: String) -> [String] in
+            [
+                "let foo = \(sign)10_0",
+                "let foo = \(sign)1000",
+                "let foo = \(sign)1000e2",
+                "let foo = \(sign)1__000",
+                "let foo = \(sign)1.0001",
+                "let foo = \(sign)1_000_000.000000_1",
+                "let foo = \(sign)1000000.000000_1"
+            ]
+        }
+    }
+
+    static func triggeringExamples() -> [String] {
```

Comment:
Basically if the value doesn't change across invocations, it should be a static or global let, unless the memory overhead of preserving that value across invocations is too high. Otherwise, that code will always be run even though the result won't change.